mycroft/stt/__init__.py

```diff
@@ -101,13 +107,38 @@ def execute(self, audio, language=None):
         language = language or self.lang
         return self.api.stt(audio.get_flac_data(), language, 1)[0]
 
+#KaldiSTT added. Unsure which class to use BasicSTT, STT or TokenSTT
+#The problem is that with "module": "kaldi" in the config file, mycroft still falls back to MycroftSTT. "update" in config has to be set to false
+
+class KaldiSTT(STT):
+    def __init__(self):
+        super(KaldiSTT, self).__init__()
+        #self.api = STTApi() necessary?
+
+    def execute(self, audio, language=None):
+        language = language or self.lang
+        
+        port_kaldi = 8081 #default port        
+        config = ConfigurationManager.get().get("stt", {})
+        port_kaldi = config.get("port")
+        print("Port for localhost Kaldi Server:", port_kaldi)
+        
+        #the kaldigstserver has to run at localhost on the defined port in the config file under stt "port": xxxx
+        kaldi_server_response = requests.post("http://localhost:%s/client/dynamic/recognize" % port_kaldi, data=audio.get_wav_data())        
```

Comment:
As suggested above, use an URI in your kalfi configuration and you don't have to bind to a fix host and port.

Then, you'd have:
```
response = requests.post(config.get("uri"), data=audio.get_wav_data())        
```

NOTE: This lonely line would replace your code from line 120 to 127.